scripts/build-maven-deploy-maven-central.bash

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+# Installs (most) maven dependencies and processes them.
+
+set -exo pipefail
+
+OPTIONS=":U:P:G:R:F:S:Z:R:"
+
+usage()
+{
+	set +x
+	USAGE=$(cat <<- USAGE_STRING
+		Usage: $(basename "${0}") [OPTION]...
+
+		Required Arguments:
+		  -U location	OSSRH_JIRA_ID
+		  -P location	OSSRH_JIRA_PASSWORD
+		  -T location	Optional maven settings security  file. Default: \`./.mvn/settings-security.xml\`
+			-K id user for signing release
+
+		Optional Arguments:
+		  -R location Optional alternative deployment repository. Default: \`\`
+		  -F location Optional pom.xml file location. Default: \`pom.xml\`
+		  -S location	Optional maven settings file. Default: \`./.mvn/settings.xml\`
+		  -Z location	Optional maven cache directory. Default: \`./.m2\`
+
+
+		USAGE_STRING
+	)
+
+	echo "${USAGE}"
+
+	set -x
+}
+# Detect `--help`, show usage and exit
+
+for var in "$@"; do
+	if [ "${var}" == '--help' ]; then
+		usage
+		exit 0
+	fi
+done
+
+while getopts "${OPTIONS}" option
+do
+	case "${option}" in
+        # Required private signing key
+        G  ) GPG_KEY_SIGNING_ID="${OPTARG}";;
+        U  ) OSSRH_JIRA_ID="${OPTARG}";;
+        P  ) OSSRH_JIRA_PASSWORD="${OPTARG}";;
+        # Optional
+        R  ) ALT_DEPLOYMENT_REPOSITORY="${OPTARG}";;
+        F  ) POM_FILE="${OPTARG}";;
+        S  ) SETTINGS_LOCATION="${OPTARG}";;
+        Z  ) M2_LOCATION="${OPTARG}";;
+
+        \? ) echo "Unknown option: -${OPTARG}" >&2; exit 1;;
+        :  ) echo "Missing option argument for -${OPTARG}" >&2; exit 1;;
+        *  ) echo "Unimplemented option: -${option}. This is probably unintended." >&2; exit 1;;
+    esac
+done
+
+if [ -z "${M2_LOCATION}" ]; then
+	M2_LOCATION="./.m2"
+fi
+
+MAVEN_OPTIONS=" -Dmaven.test.skip=true -Dmaven.repo.local=${M2_LOCATION}  --no-transfer-progress "
+
+if [ "${SETTINGS_LOCATION}" ]; then
+	MAVEN_OPTIONS+=" --settings ${SETTINGS_LOCATION} "
+fi
+
+if [ "${OSSRH_JIRA_ID}" ]; then
```

Comment:
A check is needed for the case where we have only the user or the password but not both

Reply:
John , why u need this check

Reply:
Because if by mistake only one is provided we'll have to debug this from the pipelines and I'd rather not. Having a check in place will save us future headaches.